framework/util/show_coverage.pl

```diff
@@ -109,7 +115,12 @@ =head1 DESCRIPTION
             }
     }
 
-    printf("\nNumber tests: %d,  %d failed\n", $test_count, $tot_fail);
+    if ($expected_test_count < $test_count) {
+        die "More test results than expected!";
+    }
+    $tot_fail = $expected_test_count - $test_count + $tot_fail;
```

Comment:
I found this line confusing; clearer would be
```$tot_fail += $expected_test_count;```
But, why is this necessary?  It confused me that tot_fail is computed in two different ways.
Maybe it would be better to compute passing tests, failing tests, and then report how many tests aren't found at all.